### ISSUE TITLE
Rails/Date Date#to_time false positive

### ISSUE THREAD

**Issue by savef** [bbatsov/rubocop#3312]
## Expected behavior

I should be able to call `to_time` on strings without causing an offence.
## Actual behavior

`"2016-07-12 14:36:31 +0100".to_time` and `variable_or_method.to_time` both say _do not use `to_time` on Date objects, because they know nothing about the time zone in use_. This is of course not true in the case of the string literal being parsed. It may or may not be true of a variable or method call, but there's no way for RuboCop to make the distinction so it shouldn't attempt to.
## RuboCop version

```
$ rubocop -V
0.41.2 (using Parser 2.3.1.2, running on ruby 2.3.0 x86_64-linux)
```

**Comment by savef** [comment]
@bbatsov, thoughts? Maybe just disable the cop by default?

**Comment by bbatsov** [comment]
This cop is tricky, but useful. We should probably add a check for a string literal receiver and other cases can be handled by disabling the cop inline. Half the performance and Rails cops are problematic for similar reasons, but people seems to find them more useful than annoying.

**Comment by jcoyne** [comment]
> seems to find them more useful than annoying

I find those that give false positives to be more annoying than useful.
